### ISSUE TITLE
Ability to provide ssh keys for cloning

### ISSUE THREAD

**Issue by torstenek** [docker/dev-environments#18]
**Tell us about your request**
A clear and concise description of what you want to happen or the change you would like to see

**Which service(s) is this request for?**
Let us know which product(s) you want this for?

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**

**Are you currently working around the issue?**
A clear and concise description of any alternative solutions or features you've considered or are using today.

**Additional context**

**Comment by rumpl** [comment]
Hey @torstenek, can you explain more what you mean by this please? When cloning over SSH we talk to the ssh-agent so any key that is loaded in there can and will be used during cloning.

**Comment by torstenek** [comment]
Cloning a private repo with https to the dev container works fine.
Using the ssh url fails with 'denied public key'
Works fine in a shell.

I'm probably missing something?

- T

**Comment by rumpl** [comment]
[We updated the docs the other day](https://docs.docker.com/desktop/dev-environments/#start-a-single-container-dev-environment), the important part is `When cloning a Git repository using SSH, ensure you’ve added your SSH key to the ssh-agent. To do this, open a terminal and run ssh-add <path to your public ssh key>.`

Can you check and see if that works for you?

**Comment by torstenek** [comment]
That was it. Silly me. That pro tip should be made more prominent. I think
others who are in TL;DR-mode will encounter this too :)

**Comment by rumpl** [comment]
I agree, that's why we decided to add a check if there are any keys in the agent when you give us a `git://` url.

So it should be more prominent that way :)

**Comment by torstenek** [comment]
Works as expected

**Comment by electrocnic** [comment]
This does not work for me as expected:
I started an ssh-agent in the background inside WSL as well as on the Windows host and on both I added my private key with ssh-add.
Still the docker desktop client gives me the error "When cloning a Git repository using SSH, ensure you’ve added your SSH key to the ssh-agent. To do this, open a terminal and run ssh-add <path to your public ssh key>." when I try to provide a "git@" URL when creating a dev-environment.

This is on Windows 11.
